src/mbed_cloud/certificates/certificates.py

```diff
@@ -146,7 +144,7 @@ def add_certificate(self, name, type, certificate_data, signature, **kwargs):
 
         kwargs.update({'certificate_data': certificate_data})
         certificate = Certificate._create_request_map(kwargs)
-        if not certificate.get('enrollment_mode'):
+        if not certificate.get('enrollment_mode') and signature:
```

Comment:
Can this just be removed if it will never be true?

Reply:
It is still a parameter and technically only deprecated so I send it if provided.

Reply:
ack